src/app/data-planes/components/DataPlaneConfig.vue

```diff
@@ -0,0 +1,38 @@
+<template>
+  <div>
+    <DataSource
+      v-slot="{ data, error }: DataplaneSource"
+      :src="`/meshes/${props.mesh}/dataplanes/${props.dataPlane}`"
+    >
```

Comment:
I think there was a point about not doing `DataSource`  calls in reusable components. The main reason was that as soon as you start doing this you stop controlling what does requests and you slowly fallback to what we used to do (Pull half the world on startup multiple times without much understanding and deduping)

Reply:
That’s a fair point, but it has a price, too: this component, as of this PR, is used 5 times. We’d be duplicating this component’s content more or less exactly in those 5 uses. We have to decide what’s better here.

@johncowen Do you have any opinions here?

Reply:
My opinion is we should _try_ not to use DataSource in re-usable/presentational components at all. We should pass the data down. As a general rule DataSource should only appear in route/view components.

Half the reason why DataSource is made to be a component it's then easier to compose it with other reusable components. Just like an `img`.

Overall, I'd want to look at this PR a little better first though. 🤔 But that being said, isn't there a comment in this PR somewhere about starting out our summary views to only use data from the listing response? If so, that would mean this component would go altogether for the moment right? Meaning we don't have to discuss bit here? Removing the component also means we don't have wrap this code up in a further component. For the moment it can stay in the route view, and as @lahabana says that avoids us putting this DataSource in a 'reusable' component.

Let me know if I misunderstood tho

Reply:
That's my understanding as well

Reply:
I generally moved all the things making up the tray views into those tray views again.